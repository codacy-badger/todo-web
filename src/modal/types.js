export const CONFIRMATION = "confirmation";
export const SELECT = "select";
export const INPUT = "input";
