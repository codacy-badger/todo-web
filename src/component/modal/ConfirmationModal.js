import React from "react";
import "component/modal/ConfirmationModal.sass";
import { withTranslation } from "react-i18next";

function ConfirmationModal({
  title = "Title",
  text = "Modal description text",
  visible = false,
  onClose = () => {},
  onConfirm = () => {},
  t
}) {
  return visible ? (
    <div className="confirmation-modal-component" onClick={() => onClose()}>
      <div className="modal-body" onClick={e => e.stopPropagation()}>
        <h2 className="modal-title">{title}</h2>
        <div className="modal-description">{text}</div>
        <div className="modal-confirmation-button">
          <span
            className="confirmation-text"
            onClick={() => {
              onConfirm();
              onClose();
            }}
          >
            {t("modal.confirmation")}
          </span>
        </div>
      </div>
    </div>
  ) : null;
}

export default withTranslation()(ConfirmationModal);
