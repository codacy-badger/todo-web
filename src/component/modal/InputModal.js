import React from "react";
import { withTranslation } from "react-i18next";
import "component/modal/InputModal.sass";

function InputModal({
  title = "Title",
  text = "Input modal description text",
  visible = false,
  onClose = () => {},
  onConfirm = () => {},
  t
}) {
  const modalInput = React.createRef();
  return visible ? (
    <div className="confirmation-modal-component" onClick={() => onClose()}>
      <div className="modal-body" onClick={e => e.stopPropagation()}>
        <h2 className="modal-title">{title}</h2>
        <div className="modal-description">{text}</div>
        <input ref={modalInput} className="modal-input" autoFocus />
        <div className="modal-confirmation-button">
          <span
            className="confirmation-text"
            onClick={() => {
              onConfirm(modalInput.current.value);
              onClose();
            }}
          >
            {t("modal.confirmation")}
          </span>
        </div>
      </div>
    </div>
  ) : null;
}

export default withTranslation()(InputModal);
