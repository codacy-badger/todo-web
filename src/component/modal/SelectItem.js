import React from "react";
import "component/modal/SelectItem.sass";

function SelectItem({
  image = null,
  text = "select item",
  onClick = () => {}
}) {
  const img = image ? (
    <img className="image" src={image} alt="select-item" />
  ) : null;
  return (
    <div className="select-item-component" onClick={onClick}>
      {img}
      <span className="text">{text}</span>
    </div>
  );
}

export default SelectItem;
