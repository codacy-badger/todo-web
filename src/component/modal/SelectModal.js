import React from "react";
import "component/modal/SelectModal.sass";

function SelectModal({
  title = "Title",
  text = "Modal description text",
  visible = false,
  onClose = () => {},
  children
}) {
  return visible ? (
    <div className="select-modal-component" onClick={() => onClose()}>
      <div className="modal-body" onClick={e => e.stopPropagation()}>
        <h2 className="modal-title">{title}</h2>
        <div className="modal-description">{text}</div>
        <div className="modal-selection">{children}</div>
      </div>
    </div>
  ) : null;
}

export default SelectModal;
