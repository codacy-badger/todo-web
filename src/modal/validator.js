import { CONFIRMATION, SELECT, INPUT } from "modal/types";

export function validateModalData(modal) {
  const { type = CONFIRMATION } = modal;
  switch (type) {
    case SELECT:
      return validateSelectModalData(modal);
    case INPUT:
      return validateInputModalData(modal);
    default:
      return validateConfirmationModalData(modal);
  }
}

function validateConfirmationModalData({
  visible = true,
  title = "Title",
  text = "Description",
  onConfirm = () => {}
}) {
  return {
    type: CONFIRMATION,
    visible,
    title,
    text,
    onConfirm
  };
}

function validateSelectModalData({
  visible = true,
  title = "Title",
  text = "Description",
  options = []
}) {
  return {
    type: SELECT,
    visible,
    title,
    text,
    options
  };
}

function validateInputModalData({
  visible = true,
  title = "Title",
  text = "Description",
  onConfirm = () => {}
}) {
  return {
    type: INPUT,
    visible,
    title,
    text,
    onConfirm
  };
}
