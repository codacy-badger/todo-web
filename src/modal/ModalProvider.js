import React from "react";
import ConfirmationModal from "component/modal/ConfirmationModal";
import SelectModal from "component/modal/SelectModal";
import InputModal from "component/modal/InputModal";
import { CONFIRMATION, SELECT, INPUT } from "modal/types";
import { validateModalData } from "modal/validator";

export const ModalContext = React.createContext();

class ModalProvider extends React.Component {
  state = {
    type: CONFIRMATION,
    visible: false
  };

  openModal = modal => {
    const modalData = validateModalData(modal);
    this.setState(modalData);
  };

  closeModal = () => {
    this.setState({
      visible: false
    });
  };

  mountModalComponent() {
    switch (this.state.type) {
      case SELECT:
        return <SelectModal />;
      case INPUT:
        return <InputModal />;
      default:
        return <ConfirmationModal />;
    }
  }

  render() {
    return (
      <ModalContext.Provider
        value={{
          modal: this.state,
          openModal: this.openModal,
          closeModal: this.closeModal
        }}
      >
        {this.mountModalComponent()}
        {this.props.children}
      </ModalContext.Provider>
    );
  }
}

export default ModalProvider;
