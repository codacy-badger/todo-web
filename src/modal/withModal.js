import React from "react";
import { ModalContext } from "modal/ModalProvider";

export function withModal(Component) {
  return function Wrapper(props) {
    return (
      <ModalContext.Consumer>
        {modalContext => (
          <Component
            {...props}
            modal={modalContext.modal}
            openModal={modalContext.openModal}
            closeModal={modalContext.closeModal}
          />
        )}
      </ModalContext.Consumer>
    );
  };
}
