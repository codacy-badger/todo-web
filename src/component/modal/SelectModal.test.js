import React from "react";
import ModalProvider from "modal/ModalProvider";
import renderer from "react-test-renderer";
import { SELECT } from "modal/types";

const TEST_MODAL_SELECT = {
  type: SELECT,
  visible: true,
  title: "New list",
  text: "Choose name for your new todo list:"
};

describe("Select modal", () => {
  test("should render input modal", () => {
    const component = renderer
      .create(<ModalProvider init={TEST_MODAL_SELECT} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
