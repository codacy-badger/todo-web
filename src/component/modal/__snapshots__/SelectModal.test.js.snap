// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select modal should render input modal 1`] = `
<div
  className="select-modal-component"
  onClick={[Function]}
>
  <div
    className="modal-body"
    onClick={[Function]}
  >
    <h2
      className="modal-title"
    >
      New list
    </h2>
    <div
      className="modal-description"
    >
      Choose name for your new todo list:
    </div>
  </div>
</div>
`;
